recipes/ctpp_censustract_variables/build.sql

```diff
@@ -0,0 +1,197 @@
+            lineno AS variable,
+            (CASE
+                WHEN lineno = '1' THEN 'trans_total'
+                WHEN lineno = '2' THEN 'trans_auto_solo'
+                WHEN lineno = '3' THEN 'trans_auto_2'
+                WHEN lineno = '4' THEN 'trans_auto_3'
+                WHEN lineno = '5' THEN 'trans_auto_4'
+                WHEN lineno = '6' THEN 'trans_auto_5_or_6'
+                WHEN lineno = '7' THEN 'trans_auto_7_or_more'
+                WHEN lineno = '8' THEN 'trans_public_bus'
+                WHEN lineno = '9' THEN 'trans_public_streetcar'
+                WHEN lineno = '10' THEN 'trans_public_subway'
+                WHEN lineno = '11' THEN 'trans_public_rail'
+                WHEN lineno = '12' THEN 'trans_public_ferry'
+                WHEN lineno = '13' THEN 'trans_bicycle'
+                WHEN lineno = '14' THEN 'trans_walk'
+                WHEN lineno = '15' THEN 'trans_taxi'
+                WHEN lineno = '16' THEN 'trans_motorcycle'
+                WHEN lineno = '17' THEN 'trans_other'
+                WHEN lineno = '18' THEN 'trans_home'
+            END) AS mode,
+            REPLACE(est, ',', '') AS value,
+            REPLACE(replace(moe, ',' ,''), '+/-', '') as moe
+        FROM ctpp_mode_split_ny."2012_2016"
+        WHERE LEFT(split_part(geoid, 'US', 2),5) in 
+            ('36061', '36047', '36005', '36081', '36085')
+            AND LENGTH(geoid) = 18
+    ) a
+    GROUP BY geoid
+);
+
+CREATE TEMP TABLE tmp as (
+   WITH 
+    _MOE AS (
+        SELECT
+            a.geoid,
+            b.*, 
+            SQRT(
+                COALESCE(POWER(b.trans_auto_2, 2), 0) +
+                COALESCE(POWER(b.trans_auto_3, 2), 0) +
+                COALESCE(POWER(b.trans_auto_4, 2), 0) +
+                COALESCE(POWER(b.trans_auto_5_or_6, 2), 0) +
+                COALESCE(POWER(b.trans_auto_7_or_more, 2), 0)
+            ) as trans_auto_carpool_total,
+
+            SQRT(
+                COALESCE(POWER(b.trans_auto_solo, 2), 0) +
+                COALESCE(POWER(b.trans_auto_2, 2), 0) +
+                COALESCE(POWER(b.trans_auto_3, 2), 0) +
+                COALESCE(POWER(b.trans_auto_4, 2), 0) +
+                COALESCE(POWER(b.trans_auto_5_or_6, 2), 0) +
+                COALESCE(POWER(b.trans_auto_7_or_more, 2), 0)
+            ) as trans_auto_total,
+
+            SQRT(
+                COALESCE(POWER(b.trans_public_bus, 2), 0) +
+                COALESCE(POWER(b.trans_public_streetcar, 2), 0) +
+                COALESCE(POWER(b.trans_public_subway, 2), 0) +
+                COALESCE(POWER(b.trans_public_rail, 2), 0) +
+                COALESCE(POWER(b.trans_public_ferry, 2), 0)
+            ) as trans_public_total
+        FROM RECODE a, 
+            -- pivot jsonb to columns (key -> field name, value -> field value), 
+            -- this step is needed because not all tracts have all modes of travel
+            -- NULLs will be filled with 0s and calculated as 0s
```

Comment:
Do we know why certain tracts have missing data? Typically in survey data 0 means people were surveyed and none responded "yes." There are 0 estimates in the raw data, which makes me think that if the missing records were 0, they would have been included.  Is it possible to fill NULLS with 0 just for calculating the sum but not when outputting the final table?

Reply:
The raw data doesn't have all 18 modes for all census tracts. we should probably ask labs for comments, because if we remove the 0 records, it might break the front end.